src/workerd/io/io-context.c++

```diff
@@ -1109,22 +1116,18 @@ void IoContext::runImpl(Runnable& runnable, bool takePendingEvent,
   });
 }
 
-static constexpr auto kAsyncIoErrorMessage =
-    "Disallowed operation called within global scope. Asynchronous I/O "
-    "(ex: fetch() or connect()), setting a timeout, and generating random "
-    "values are not allowed within global scope. To fix this error, perform this "
-    "operation within a handler. "
-    "https://developers.cloudflare.com/workers/runtime-apis/handlers/";
-
 IoContext& IoContext::current() {
   if (threadLocalRequest == nullptr) {
     v8::Isolate* isolate = v8::Isolate::TryGetCurrent();
-    if (isolate == nullptr) {
-      KJ_FAIL_REQUIRE("there is no current request on this thread");
-    } else {
-      isolate->ThrowError(jsg::v8StrIntern(isolate, kAsyncIoErrorMessage));
-      throw jsg::JsExceptionThrown();
-    }
+    KJ_REQUIRE(isolate != nullptr, "there is no current request on this thread");
+    jsg::Lock& js = jsg::Lock::from(isolate);
+    js.throwException(js.strIntern(
+        "Disallowed operation called within global scope. Asynchronous I/O "
+        "(ex: fetch() or connect()), setting a timeout, and generating random "
+        "values are not allowed within global scope. To fix this error, perform this "
+        "operation within a handler. "
+        "https://developers.cloudflare.com/workers/runtime-apis/handlers/"_kj
+    ));
```

Comment:
This part of the change is not technically necessary but it makes this bit more compact/cleaner so kept it in.